### ISSUE TITLE
Compilation error (BB #457)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#457]
This issue was migrated from bitbucket.
**Original Reporter:** funn3r
**Original Date:** 05.05.2013 11:35:57 GMT+0000
**Original Priority:** critical
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/457

<hr>

Fixed

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 06.01.2014 16:46:28 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/457/#comment-7607984

<hr>

Here is diff:

```
 #include "World.h"
 #include "Util.h"

+
 int PetAI::Permissible(const Creature *creature)
 {
     if (creature->isPet())
@@ -162,7 +163,7 @@ void PetAI::UpdateAI(const uint32 diff)
             // Fix to allow pets on STAY to autocast
             if (me->getVictim() && _CanAttack(me->getVictim()) && spell->CanAutoCast(me->getVictim()))
             {
-                targetSpellStore.push_back(std::make_pair<Unit*, Spell*>(me->getVictim(), spell));
+                targetSpellStore.push_back(std::make_pair(me->getVictim(), spell));
                 continue;
             }
             else
@@ -178,7 +179,7 @@ void PetAI::UpdateAI(const uint32 diff)

                     if (spell->CanAutoCast(Target))
                     {
-                        targetSpellStore.push_back(std::make_pair<Unit*, Spell*>(Target, spell));
+                        targetSpellStore.push_back(std::make_pair(Target, spell));
                         spellUsed = true;
                         break;
```

I don't think that explicitly setting template does some harm...  
Could you post your error message?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 19.01.2014 23:52:15 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/457/#comment-7813038

<hr>

Noone else has complained about this issue. Do you think I'd be safe to close this Zaffy? I am curious what OS he was using to stumble upon such an issue.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 20.01.2014 06:40:37 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/457/#comment-7818671

<hr>

It looks like [this commit](https://bitbucket.org/oregon/oregoncore/commits/326480de68f73ef7848cc3af306e701e85985cc3) has alredy fixed this issue (it it was one).

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Yooki
**Original Date:** 20.01.2014 12:40:08 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/457/#comment-7823651

<hr>

Resolved with commit 326480d by Je7.
